api_common.go

```diff
@@ -103,6 +94,62 @@ func (h *Headscale) generateMapResponse(
 		},
 	}
 
+	// Send only deltas if this is the next map response in a streamed sequence
```

Comment:
Can you split this out to a function and add some tests?

nil (full update), node added, node removed, no change.

Reply:
please write table driven tests, and do not take inspiration from our old terrible tests :)

Reply:
Sounds good... can you point me in the right direction in terms of where to put this test and and preferred framework for table driven tests? I've only written table driven tests in JS

Reply:
You don't need a framework, but a structure like this should suffice https://github.com/kradalby/headscale/blob/acl-refactor/acls_test.go#L1698C1-L1834